### PR TITLE
Enable category groups on map filter dropdown

### PR DESCRIPTION
**Pull request by davea**
Adds category groups on the map filter UI:
image

Fixes https://github.com/mysociety/fixmystreet/issues/2702

### CODE REVIEW

File: t/app/controller/reports.t [hunk 1]
```diff
@@ -18,6 +18,9 @@ my $body_west_id = $mech->create_body_ok(2504, 'Westminster City Council')->id;
 my $body_fife_id = $mech->create_body_ok(2649, 'Fife Council')->id;
 my $body_slash_id = $mech->create_body_ok(10000, 'Electricity/Gas Council')->id;
 
+my $pothole = $mech->create_contact_ok(body_id => $body_edin_id, category => 'Potholes', email => 'potholes@example.org');
+my $graffiti = $mech->create_contact_ok(body_id => $body_west_id, category => 'Graffiti', email => 'graffiti@example.org');
+my $flytipping = $mech->create_contact_ok(body_id => $body_fife_id, category => 'Flytipping', email => 'flytipping@example.org');
```
Comment (struan) [inline t/app/controller/reports.t]: I'm not sure I understand what adding these lines is doing?

File: t/app/controller/reports.t [hunk 2]
```diff
@@ -128,6 +131,7 @@ $mech->content_contains('2,3,4,4');
 FixMyStreet::override_config {
     ALLOWED_COBRANDS => 'fixmystreet',
     MAPIT_URL => 'http://mapit.uk/',
+    COBRAND_FEATURES => { category_groups => { fixmystreet => 1 } },
```
Comment (struan) [inline t/app/controller/reports.t]: likewise, why add this?

File: t/app/controller/around.t [hunk 3]
```diff
@@ -232,7 +232,11 @@ subtest 'check category, status and extra filtering works on /around' => sub {
 
     # Create one open and one fixed report in each category
     foreach my $category ( @$categories ) {
-        $mech->create_contact_ok( category => $category, body_id => $body->id, email => "$category\@example.org" );
+        my $contact = $mech->create_contact_ok( category => $category, body_id => $body->id, email => "$category\@example.org" );
+        if ($category ne 'Pothole') {
+            $contact->set_extra_metadata(group => ['Environment']);
+            $contact->update;
+        }
```
Comment (struan) [inline t/app/controller/around.t]: I don't understand what the changes in this file are for.

File: t/app/controller/reports.t (hunk 1 above)
Comment (dracos) [reply t/app/controller/reports.t]: Nothing I think, actually, but it seemed odd to be creating reports just underneath that didn't have any matching categories existing for them. I can take them back out.

File: t/app/controller/around.t (hunk 3 above)
Comment (dracos) [reply t/app/controller/around.t]: This is so the fetched around page is loading the filter with category groups in it. I should perhaps have checked for that being in the output!

**Pull request merged**